datalad/crawler/main.py

```diff
@@ -232,7 +233,8 @@ def page2annex(self, existing='check', dry_run=False, cache=False):
                         incoming_filename, incoming_downloaded, incoming_updated, downloaded_size = \
                           download_url_to_incoming(href_full, incoming_annex.path,
                                        join(repo_sectiondir, href_dir),
-                                       db_incoming=db_incoming, dry_run=self.runner.dry,  # TODO -- use runner?
+                                       db_incoming=db_incoming,
+                                       dry_run=self.runner.protocol.__class__.__name__ == 'DryRunProtocol',  # TODO -- use runner?
```

Comment:
Re isinstance: *facepalm* ;) But yes - this is to be RF'ed anyway.

Re using dry_run: May be - I just didn't want to discover that code at all, just do the RF'ing. So I simply didn't know what context I was in. :)

Re side-note: I don't think so (yet). Because I use the Runner mostly from within the repo-classes. And what protocol to use should be determined outside of them, in order to run a handle-function or a datalad-command with a certain protocol. So decorating the calls to be protocolled would make things more complicated, I guess.